src/SHA.jl

```diff
@@ -2,25 +2,34 @@ module SHA
 
 using Compat
 
-export sha224, sha256, sha384, sha512
+export sha1, sha224, sha256, sha384, sha512
 
 include("constants.jl")
-include("base_functions.jl")
 include("types.jl")
+include("base_functions.jl")
+include("sha1.jl")
 include("sha2.jl")
+include("common.jl")
 
-_shasum(ctx::SHA_CTX, io::IO) =
-    (update!(ctx, readbytes(io)); bytes2hex(digest!(ctx)))
 
-for (f, ctx) in [(:sha224, :SHA224_CTX),
+# Create data types and convenience functions for each hash implemented
+for (f, ctx) in [(:sha1, :SHA1_CTX),
+				 (:sha224, :SHA224_CTX),
                  (:sha256, :SHA256_CTX),
                  (:sha384, :SHA384_CTX),
                  (:sha512, :SHA512_CTX)]
     @eval begin
-        $f(io::IO) = _shasum($ctx(), io)
+    	# Allows e.g. sha256(open("test.txt"))
```

Comment:
I rode the train from Seattle down to Portland and back.  Very pleasant transportation method, went to an Intel academic event as some of my research is funded by them through their center for pervasive computing.